runtime/common/src/test/kotlin/kotlinx/serialization/json/JsonTestBase.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json
+
+import kotlinx.serialization.*
+import kotlinx.serialization.context.*
+import kotlinx.serialization.json.internal.*
+import kotlinx.serialization.json.serializers.*
+import kotlin.test.*
+
+abstract class JsonTestBase {
+    protected val strict = Json()
+    protected val unquoted = Json(unquoted = true)
+    protected val nonstrict = Json(strictMode = false)
+
+    internal inline fun <reified T : Any> Json.stringify(value: T, useStreaming: Boolean): String {
+        val serializer = context.getOrDefault(T::class)
+        return stringify(serializer, value, useStreaming)
+    }
+
+    internal fun <T> Json.stringify(serializer: SerializationStrategy<T>, value: T, useStreaming: Boolean): String {
+        return if (useStreaming) {
+            stringify(serializer, value)
+        } else {
+            val tree = writeJson(value, serializer)
+            // kotlinx.serialization/issues/277
+            stringify(JsonElementSerializer, tree)
+        }
+    }
+
+    @ImplicitReflectionSerializer
+    inline fun <reified T : Any> Json.stringify(list: List<T>, useStreaming: Boolean): String {
+        return if (useStreaming) {
+            stringify(list)
+        } else {
+            stringify(context.getOrDefault(T::class).list, list)
```

Comment:
Looks strange to me. Why switch streaming/non-streaming should wrap in list additionally? Did you want `JsonArraySerializer` or `JsonElementSerializer.list`?

Reply:
No, I wanted particular public extensions (ones from `SerialImplicits`) to be tested.
I will leave a comment here